Sources/Subs-Attachments.php

```diff
@@ -1320,7 +1307,13 @@ function loadAttachmentContext($id_msg, $attachments)
 
 	// Do we need to instigate a sort?
 	if ($have_unapproved)
-		usort($attachmentData, 'approved_attach_sort');
+		uksort($attachmentData, function($a, $b)
```

Comment:
Why did this change from a `usort()` to a `uksort()`?

Reply:
the keys are required to remain unchanged for https://github.com/SimpleMachines/SMF2.1/pull/5574/files#diff-7773b8079c8a8f4db3a661335b297c20R983  otherwise you get undefined indexes while using the attach bbcode on unapproved attachments.

Reply:
does that mean this PR should fix #5134 ?

Reply:
No idea.

Reply:
But [uksort](https://www.php.net/manual/en/function.uksort.php) sorts by comparing the keys instead of the values. Since the goal is to preserve the keys while sorting by comparing content, I think [uasort](https://www.php.net/manual/en/function.uasort.php) is the function you want.